numba_dppy/tests/test_debuginfo.py

```diff
@@ -142,3 +142,83 @@ def foo(arr):
         for tag in ir_tags:
             got = make_check(kernel_ir, tag)
             assert expect == got
+
+
+def test_debug_flag_generates_ir_with_debuginfo_for_func(offload_device):
+    """
+    Check debug info is emitting to IR if debug parameter is set to True
+    """
+
+    if skip_test(offload_device):
+        pytest.skip()
+
+    if offload_device in "level_zero:gpu:0":
+        pytest.xfail("Failing compilation: SyclProgramCompilationError")
+
+    @dppy.func(debug=True)
+    def func_sum(a, b):
+        result = a + b
+        return result
+
+    @dppy.kernel(debug=True)
+    def data_parallel_sum(a, b, c):
+        i = dppy.get_global_id(0)
+        c[i] = func_sum(a[i], b[i])
+
+    ir_tag_func_sum = r'\!DISubprogram\(name: "test_debug_flag_generates_ir_with_debuginfo_for_func.<locals>.func_sum"'
+    ir_tag_data_parallel_sum = r'\!DISubprogram\(name: "test_debug_flag_generates_ir_with_debuginfo_for_func.<locals>.data_parallel_sum"'
+    ir_tags = (ir_tag_func_sum, ir_tag_data_parallel_sum)
+
+    with dpctl.device_context(offload_device) as sycl_queue:
+        sig = (
+            types.float32[:],
+            types.float32[:],
+            types.float32[:],
+        )
+        kernel_ir = get_kernel_ir(sycl_queue, data_parallel_sum, sig, debug=True)
+
+    for tag in ir_tags:
+        got = make_check(kernel_ir, tag)
+        assert got == True
+
+
+def test_env_var_generates_ir_with_debuginfo_for_func(offload_device):
+    """
+    Check debug info is emitting to IR if NUMBA_DPPY_DEBUGINFO is set to 1
+    """
+
+    if skip_test(offload_device):
+        pytest.skip()
+
+    if offload_device in "level_zero:gpu:0":
+        pytest.xfail("Failing compilation: SyclProgramCompilationError")
+
+    @dppy.func
+    def func_sum(a, b):
+        result = a + b
+        return result
+
+    @dppy.kernel
+    def data_parallel_sum(a, b, c):
+        i = dppy.get_global_id(0)
+        c[i] = func_sum(a[i], b[i])
+
+    ir_tag_func_sum = r'\!DISubprogram\(name: "test_env_var_generates_ir_with_debuginfo_for_func.<locals>.func_sum"'
+    ir_tag_data_parallel_sum = r'\!DISubprogram\(name: "test_env_var_generates_ir_with_debuginfo_for_func.<locals>.data_parallel_sum"'
+    ir_tags = (ir_tag_func_sum, ir_tag_data_parallel_sum)
+
+    dppy.compiler.DEBUGINFO = 1
```

Comment:
All config variables should be accessed via `config` module.

Reply:
If we change the variables from the config module it won't work. Because they are read 1 time before we import numba-dppy

Reply:
Please, explain in details. I think it is problem in imports (create ticket in this way) and we do not have to do workaround.

Reply:
Created https://github.com/IntelPython/numba-dppy/issues/442

Reply:
#424 does not answer my question about using `config` module instead of using `compiler` module.

Reply:
In the config module, the behavior is the same as with environment variables, and the variable from the compiler can be changed after importing dppy

Reply:
@1e-to so we have TWO different `DEBUGINFO` flags? One in `config` and one in `compiler`?

Reply:
It doesn't looks so...

Reply:
Fixed.